### PR TITLE
Add cmake option `BUILD_APP_LIB`

### PR DESCRIPTION
**Pull request by alvin1221**

### CODE REVIEW

**Review by lee-emqx — APPROVED** [review]

**Review by JaylinYu — APPROVED** [review]
Check IPC & pid

**Comment by JaylinYu** [comment]
plz fix conflicts

File: CMakeLists.txt [hunk 1]
```diff
@@ -27,6 +27,7 @@ option (BUILD_ZMQ_GATEWAY "Build zmq gateway" OFF)
 option (BUILD_VSOMEIP_GATEWAY "Build vsomeip gateway" OFF)
 option (BUILD_DDS_PROXY "Build dds proxy" OFF)
 option (BUILD_BENCH "Build nano-bench" OFF)
+option (BUILD_APP_LIB "Build library for app" OFF)
```
Comment (wanghaEMQ) [inline CMakeLists.txt:30]: What is the library for app?

**Review by wanghaEMQ — APPROVED** [review]

**Comment by wanghaEMQ** [comment]
Is there are any descriptions of the library for app?

File: CMakeLists.txt (hunk 1 above)
Comment (alvin1221) [reply CMakeLists.txt:30]: Build nanomq as a library and called by user's program

**Review by Hermann0222 — APPROVED** [review]

**Pull request merged**